cblox/include/cblox/core/submap_collection_inl.h

```diff
@@ -77,6 +77,19 @@ SubmapID SubmapCollection<SubmapType>::createNewSubmap(
   return new_ID;
 }
 
+template <typename SubmapType>
+void SubmapCollection<SubmapType>::addSubmap(
+    const typename SubmapType::Ptr& submap) {
```

Comment:
I think passing ptr ref is a bit dangerous because then the function does not become the owner of the ptr and therefore there is no guarantee that the ptr will be alive during the execution of this code.

Reply:
Hmm. Really good point. Good catch. I swapped this for pass by value and move.